tracer/src/Datadog.Trace/Debugger/Expressions/ProbeProcessor.cs

```diff
@@ -0,0 +1,268 @@
+                        var captureBehaviour = snapshotCreator.DefineSnapshotBehavior(info, ProbeInfo.EvaluateAt, HasCondition());
+                        if (captureBehaviour is CaptureBehaviour.Delayed or CaptureBehaviour.NoCapture)
+                        {
+                            // we need to delay the capture process
+                            return true;
+                        }
+
+                        break;
+                    case MethodState.ExitStart:
+                    case MethodState.ExitStartAsync:
+                        captureBehaviour = snapshotCreator.DefineSnapshotBehavior(info, ProbeInfo.EvaluateAt, HasCondition());
+                        snapshotCreator.AddScopeMember(info.Name, info.Type, info.Value, info.MemberKind);
+                        if (captureBehaviour == CaptureBehaviour.Delayed)
+                        {
+                            return true;
+                        }
+
+                        break;
+                    case MethodState.EntryEnd:
+                    case MethodState.ExitEnd:
+                    case MethodState.EndLine:
+                    case MethodState.EndLineAsync:
+                    case MethodState.ExitEndAsync:
+                        try
+                        {
+                            if (snapshotCreator.CaptureBehaviour == CaptureBehaviour.NoCapture)
+                            {
+                                return true;
+                            }
+
+                            snapshotCreator.AddScopeMember(info.Name, info.Type, info.Value, info.MemberKind);
+                            evaluationResult = GetOrCreateEvaluator(snapshotCreator.MethodScopeMembers).Evaluate();
+
+                            if (evaluationResult.Condition is false && (evaluationResult.Errors == null || evaluationResult.Errors.Count == 0))
+                            {
+                                // if the expression evaluated to false or the evaluation failed
+                                snapshotCreator.CaptureBehaviour = CaptureBehaviour.NoCapture;
+                                return true;
+                            }
+
+                            if (evaluationResult.Metric.HasValue)
+                            {
+                                LiveDebugger.Instance.SendMetrics();
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            // if the evaluation failed stop capturing
+                            Log.Error(e, "Failed to evaluate expression for probe: " + ProbeInfo.ProbeId);
+                            snapshotCreator.CaptureBehaviour = CaptureBehaviour.NoCapture;
+                            return true;
+                        }
+
+                        break;
+                    case MethodState.LogLocal:
+                    case MethodState.LogArg:
+                    case MethodState.LogException:
```

Comment:
We never set `MethodState.LogException` in the codebase. Am I missing something?